### ISSUE TITLE
Add Option for snapping tools while editing geometry/creating new features

### ISSUE THREAD

**Issue by azizaparveen** [Esri/solutions-components#1052]
### Describe requirements

We should add snapping tools and display them in the editing panel when a new feature is being created or existing features geometry is being edited. Snapping tools can help users with data quality.

I think we could add them right below the create feature and edit feature headers. We have something similar to this in Sidebar.
![image](https://github.com/user-attachments/assets/618219b2-3c98-4f36-8d22-2d1d1ba3ac61)
![image](https://github.com/user-attachments/assets/7e90807f-5c2a-4d2c-bc45-4f0b1fa8855a)
![image](https://github.com/user-attachments/assets/0dfb8c2d-e9ef-4a67-bab5-b8cdd3be7a36)

**Comment by sumitzarkar** [comment]
@azizaparveen 

- Snapping tools are part of the Editor widgets setting section
- Currently, we are hiding them using CSS overrides
- We can show this setting however, it will be shown below the message as the editor widget is rendered after the message.
- Here are a couple of screenshots of different cases
## Template picker - when multiple templates are present this page will be shown
Settings collapsed | Settings expanded
-- | --
![image](https://github.com/user-attachments/assets/e8b8c1ee-42f4-48c6-9302-834245ed8613)​ | ![image](https://github.com/user-attachments/assets/101964f4-883d-4dbc-8c71-e094cfe796b8)

## Drawing page - when we don't have multiple templates or after selecting the template this drawing page will be shown
Settings collapsed | Settings expanded
-- | --
![image](https://github.com/user-attachments/assets/91a9c3fa-37ae-42ab-8833-7963b76af643) ​ | ![image](https://github.com/user-attachments/assets/3b67fcda-294f-4861-9bd2-d7966ba9edf3) 

## Feature Form/editing page
Settings collapsed | Settings expanded
-- | --
![image](https://github.com/user-attachments/assets/c5b5cbee-2699-4d48-b9c4-8c3e04414219) ​ | ![image](https://github.com/user-attachments/assets/a8e92b45-b6e3-4556-a50a-34f6c1e108ec)

**Comment by azizaparveen** [comment]
@sumitzarkar we are not showing message in Manager at this moment, we decides to hide it for Manager. so I guess we will be fine showing the settings.

**Comment by azizaparveen** [comment]
@sarahmcdonald7 snapping tools are available now in Manager, I am wondering if we just show it when Edit geometry and/or Create feature option is enabled, When these two options are disabled it does not make much sense to show them when only attribute editing is allowed. What do you think?
![Image](https://github.com/user-attachments/assets/f82932a9-ffdd-434a-9832-b94f5eb7c242)

**Comment by sarahmcdonald7** [comment]
@azizaparveen Good point. I think if create feature and edit geometry are both disabled in the app we don't need to show the snapping tools.

**Comment by azizaparveen** [comment]
@sumitzarkar can we hide the snapping tools when Geometry edits and/or Create new features options are off.

![Image](https://github.com/user-attachments/assets/073a3098-22bd-4def-8c87-ddd9f5754d90)

**Comment by pratiksabale07** [comment]
@azizaparveen @sarahmcdonald7
 
## Cause
- This is an enhancement requested by the client.
- Previously the snapping controls were always shown.
___
## Solution
- If **Geometry Edits** is enabled in the configuration, then enable the **snapping controls** in the **edit card** (and vice versa).  
- If **Create New Features** is enabled in the configuration, then enable the **snapping controls** for feature creation (and vice versa).  
___
## Dev Checklist
Items  | Is required | Status | Comments
-- | -- | -- | --
Required NLS changes? | No   |   | 
Required Backward compatibility | No  |   | 
Required 508 | Yes  |   | 
Need Sanitize? | No  |   | 
___
## Impact Analysis Report
Impacted Areas | Comments
-- | --
Have Configuration changes? | No
Have Runtime changes? | Yes
Impact on RTL | Yes
Impacted components | - create-feature </br> - edit-card
Impact on runtime | -  If **Geometry Edits** is enabled in the configuration, the **snapping controls** in the **edit card** should be displayed.
||-  If **Create New Features** is enabled in the configuration, the **snapping controls** should be shown for feature creation
